### ISSUE TITLE
Issues with check-class-attributes and properties, as well as 'private' class variables

### ISSUE THREAD

**Issue by Midnighter** [jsh9/pydoclint#148]
I stumbled over two problems with pydoclint 0.5.3 recently (occurs on Python 3.12 but not 3.10).

`pyproject.toml`:
```toml
[tool.pydoclint]
style = "google"
arg-type-hints-in-docstring = false
check-return-types = false
check-yield-types = false
```

`mre.py`:
```py
class MinimalReproducibleExample:
    """
    Define a minimal reproducible example of check-class-attributes.

    Args:
        value: A string value.

    Attributes:
        value: A string value.

    """

    _HIDDEN = 23

    def __init__(self, value: str) -> None:
        super().__init__()
        self._value = value
        self._hidden = self._HIDDEN

    @property
    def value(self) -> str:
        return self._value
```

```text
Loading config from user-specified .toml file: pyproject.toml
Found options defined in pyproject.toml:
{'style': 'google', 'arg_type_hints_in_docstring': False, 'check_return_types': False, 'check_yield_types': False}
Skipping files that match this pattern: \.git|\.tox
mre.py
mre.py
    1: DOC603: Class `MinimalReproducibleExample`: Class docstring attributes are different from actual class attributes. (Or could be other formatting issues: https://jsh9.github.io/pydoclint/violation_codes.html#notes-on-doc103 ). Attributes in the class definition but not in the docstring: [_HIDDEN: ]. Arguments in the docstring but not in the actual class attributes: [value: ]. (Please read https://jsh9.github.io/pydoclint/checking_class_attributes.html on how to correctly document class attributes.)
```

In my opinion:

1. "Private" class attributes [`_HIDDEN`] should be ignored just like "private" instance attributes [`self._hidden`].
2. I very much think that a property method should be documented just like an attribute.

**Comment by jsh9** [comment]
Hi @Midnighter , thank you for your suggestion!

I've implemented both your suggestions into a new version, 0.5.4 ([diff](https://github.com/jsh9/pydoclint/compare/0.5.3...0.5.4)).

In 0.5.4, two new options are added:

- An option --should-document-private-class-attributes (if False, private class attributes should not appear in the docstring)
- An option --treat-property-methods-as-class-attributes (if True, `@property` methods are treated like class attributes and need to be documented in the class docstring)

**Comment by Midnighter** [comment]
Fantastic, thank you!

**Comment by kesmik** [comment]
Hey, @Midnighter , @jsh9 , I would like to know, why properties needs to be treated as class attributes?
Is it common practice any pep for that? if anybody knows please let me know, because for now I set this new parameter to false, since it looks strange for me but I am not sure.

Thanks in advance!

**Comment by jsh9** [comment]
Hi @kesmik , 

I've also been re-thinking about this over the past few days.  Oftentimes properties are not to be considered class attributes, because their values are not known unless we instantiate an object from it.

Therefore, I'll make a code change today to set the default value of this config option to False. (This will also restore backward compatibility.)

**Comment by Midnighter** [comment]
My point is that from a class interface/usage perspective, they are indistinguishable from attributes in most cases. How do you suggest to document properties?

**Comment by jsh9** [comment]
> My point is that from a class interface/usage perspective, they are indistinguishable from attributes in most cases. How do you suggest to document properties?

Before you made a suggestion, I used to treat them just as a method, and then I document them below the method signature.
